src/uwtools/config.py

```diff
@@ -752,3 +753,37 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def log_and_error(msg: str, log: Logger) -> None:
+    """
+    Will log a user-provided error message and raise a UWConfigError with the same message.
+    """
+    log.error(msg)
+    raise UWConfigError(msg)
+
+
+def print_config_section(config: dict, section_path: List[str], log: Logger) -> None:
+    """
+    Descends into the config via the given section keys, then prints the contents of the located
+    subtree as key=value pairs, one per line.
+    """
+    keys = []
+    for section in section_path:
+        keys.append(section)
+        current_path = " -> ".join(keys)
+        try:
+            subconfig = config[section]
+        except KeyError:
+            log_and_error(f"Bad config path: {current_path}", log)
```

Comment:
So, I broke this on my first attempt in a weird way. I got a `TypeError` here by accidentally passing in a string instead of a dict and it took me a while to figure out where I'd gone wrong. This is a more general question, but does it make sense to make sure that our caller is obeying the contract and actually passing in a dict for config? It looked like this:

```
Traceback (most recent call last):
  File "/Users/christina.holt/Work/workflow-tools/test_export.py", line 21, in <module>
    config.print_config_section(list_example, section, log=Logger())
  File "/Users/christina.holt/Work/workflow-tools/src/uwtools/config.py", line 776, in print_config_section
    subconfig = config[section]
                ~~~~~~^^^^^^^^^
TypeError: string indices must be integers, not 'str'
```

This also got me thinking to check what happens with my first instinct...passing in `section_path` as a string instead of a list since I only wanted one. I got essentially the same TypeError. In the case the caller provided a string, does it make sense to assume the user wanted only one section and make it a list?

Reply:
The typechecker will enforce correct argument types for internal code, so we should not be doing manual checks of argument types in library modules like this one. If you add new/test code is to `src/uwtools` source files, `make typecheck` should catch and report this kind of error. If you're interacting with the code in a REPL, you're a dev power-user and on your own, but you can e.g. type `help(uwtools.config.print_config_path)` in the REPL to see the argument names, types, docstring, etc. FWIW, I did exactly the same thing: Passed in a string value for `section_path` instead of a list.

When we have API code, we might want to to extra work there in case users are not doing their own typechecking, because they probably aren't.

Reply:
I missed the last bit of the question: I think a `str` -> `List[str]` conversion could be appropriate in an API function. For the CLI case, which is all we have now, the CLI code will certainly pass in a `List[str]` object. So I don't think we should be massaging values here.

Reply:
I'm fine with not hand-holding our power users. Just thought I'd bring it up since we were already doing some error-catching here. But, yeah, RTFM is a perfectly fine way to go. :)